code/datums/black_market_item.dm

```diff
@@ -240,6 +240,21 @@ for anyone but the person committing mass murder.
 /datum/black_market_item/tech
 	category = "Advanced Technology" 
 
+/datum/black_market_item/tech/emagged_nikita
+	name = "Emagged Nikita"
+	desc = "A rocket launcher that fires missiles that you can ride. Great for special effects in Live-Action movies and terrorist roleplay."
+	item = /obj/structure/closet/crate/secure/weapon/experimental/nikita
+	sps_chances = list(0, 20, 35)
+	stock_min = 1
+	stock_max = 1
+	cost_min = 1000
+	cost_max = 2000
+	display_chance = 80
+
+/datum/black_market_item/tech/emagged_nikita/after_spawn(var/obj/structure/closet/crate/secure/weapon/experimental/nikita/spawned, var/mob/user)
+	var/obj/item/weapon/gun/projectile/rocketlauncher/nikita/rocket_launcher = locate() in spawned
+	rocket_launcher.emagged = TRUE
```

Comment:
This is really, _really_ poor code design. What if it turns out that in allocating the nikita that there was a memory overflow error that went uncatched and the emagged variable had a null pointer? I am literally shaking right now. How could you have not seen this? This would result in a serious memory leak from which there would be no recovery. Oh my god. Why? Check if emagged has a null pointer! My god! My GOD dude. This is so incredibly stupid I cannot even fathom what kind of idiotic pre-simian macaque could neglect this critical error. Get an education. My god, fuckface.